tests/unit-tests/test_model_save_load.py

```diff
@@ -201,7 +204,78 @@ def test_sparse_embed_load_corner_cases():
     th.distributed.destroy_process_group()
     dgl.distributed.kvstore.close_kvstore()
 
+class DummySparseEmb:
+    """ Dummy class for sparse embedding
+    """
+    def __init__(
+        self,
+        shape: tuple,
+        dtype: th.dtype = th.float,
+    ):
+        self._tensor = th.zeros(shape, dtype=dtype)
+
+    @property
+    def num_embeddings(self):
+        return len(self._tensor)
+
+    def __setitem__(self, idx: th.Tensor, val: th.Tensor):
+        self._tensor[idx] = val
+
+    def __getitem__(self, idx: th.Tensor) -> th.Tensor:
+        return self._tensor[idx]
+
+@pytest.mark.parametrize("emb_size", [(140, 10), (1, 10), (13, 10)])
+@pytest.mark.parametrize("world_size", [80, 1, 12])
+def test_save_load_sparse_emb(emb_size, world_size, monkeypatch):
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        emb_path = os.path.join(tmpdirname, "path")
+
+
+        sparse_emb = DummySparseEmb(emb_size)
+        sparse_emb[th.arange(emb_size[0])] = th.rand(emb_size)
+        ntype = "test"
+        world_size = 80
+        def mock_get_world_size():
+            return world_size
+
+        for rank in range(world_size):
+            def mock_get_rank():
+                return rank
+
+            monkeypatch.setattr(utils , "get_world_size", mock_get_world_size)
+            monkeypatch.setattr(utils , "get_rank", mock_get_rank)
+            utils.save_sparse_emb(emb_path, sparse_emb, ntype)
+
+        load_sparse_emb = DummySparseEmb(emb_size)
```

Comment:
This will shadow the imported `load_sparse_emb` function. Can we use a different name to not confuse the two?